### ISSUE TITLE
Reassignment should kill type narrowing for member access expressions

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#1218]
When a member's type is narrowed but then the base expression (to the left of the ".") is reassigned, the narrowing should be eliminated.

```python
from typing import Literal

class Response:
    status: int

def func(a: bool):
    response: Response = Response()
    response.status = 0

    if a:
        response = Response()

    if response.status == 200:
        t1: Literal["int"] = reveal_type(response.status)
```

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is now addressed in pyright 1.1.91, which I just published. It will also be included in the next release of pylance.
